pkg/storage/cache.go

```diff
@@ -161,37 +160,35 @@ type CheckCacheKeyParams struct {
 // should produce the same cache key. Contextual tuple order and context parameter order is ignored,
 // only the contents are compared.
 func GetCheckCacheKey(params *CheckCacheKeyParams) (string, error) {
-	hasher := keys.NewCacheKeyHasher(xxhash.New())
-
-	key := strings.Builder{}
-	key.WriteString(SubproblemCachePrefix)
-	key.WriteString(params.StoreID)
-	key.WriteString("/")
-	key.WriteString(params.AuthorizationModelID)
-	key.WriteString("/")
-	key.WriteString(params.TupleKey.GetObject())
-	key.WriteString("#")
-	key.WriteString(params.TupleKey.GetRelation())
-	key.WriteString("@")
-	key.WriteString(params.TupleKey.GetUser())
-
-	if err := hasher.WriteString(key.String()); err != nil {
+	hasher := xxhash.New()
+
+	_, err := hasher.WriteString(
+		SubproblemCachePrefix +
+			params.StoreID +
+			"/" +
+			params.AuthorizationModelID +
+			"/" +
+			params.TupleKey.GetObject() +
+			"#" +
+			params.TupleKey.GetRelation() +
+			"@" +
+			params.TupleKey.GetUser(),
+	)
+	if err != nil {
 		return "", err
 	}
 
 	// here, and for context below, avoid hashing if we don't need to
 	if len(params.ContextualTuples) > 0 {
-		if err := keys.NewTupleKeysHasher(params.ContextualTuples...).Append(hasher); err != nil {
```

Comment:
Can we get rid of `NewTupleKeysHasher` and all its associated code? I can't find any usage of it :)